clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/UpdateInstancesAtomicOperation.groovy

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.aws.deploy.ops
+
+import com.amazonaws.services.ec2.model.ModifyInstanceAttributeRequest
+import com.netflix.spinnaker.clouddriver.aws.deploy.description.UpdateInstancesDescription
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider
+import com.netflix.spinnaker.clouddriver.aws.services.RegionScopedProviderFactory
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import org.springframework.beans.factory.annotation.Autowired
+
+class UpdateInstancesAtomicOperation implements AtomicOperation<Void> {
+  private static final String PHASE = "UPDATE"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  @Autowired
+  AmazonClientProvider amazonClientProvider
+
+  @Autowired
+  RegionScopedProviderFactory regionScopedProviderFactory
+
+  final UpdateInstancesDescription description
+
+  UpdateInstancesAtomicOperation(UpdateInstancesDescription description) {
+    this.description = description
+  }
+
+  @Override
+  Void operate(List priorOutputs) {
+    task.updateStatus PHASE, "Initializing Update Instances operation for ${description.serverGroupName}..."
+
+    def amazonEC2 = amazonClientProvider.getAmazonEC2(description.credentials, description.region, true)
+
+    def serverGroupName = description.serverGroupName
+    def credentials = description.credentials
+    def region = description.region
+
+    def regionScopedProvider = regionScopedProviderFactory.forRegion(credentials, region)
+
+    def asgService = regionScopedProvider.asgService
+    def asg = asgService.getAutoScalingGroup(serverGroupName)
+    if (!asg) {
```

Comment:
it would be worth validating that the ASG is in VPC so we can fail with a more descriptive error rather than blowing up on the API call if its an EC2 classic instance